### PR TITLE
new syntax added

### PR DESCRIPTION
**Pull request by xhudik**
I just rewrite into new Scala 3 syntax. I mean the examples are awsome, just dont know why there are in original Scala 2 format. If there is a reason - feel free to reject this pull request

### CODE REVIEW

**Review by hmf — COMMENTED** [review]
Maybe you should also remove the `new` that is nit required anymore.

**Comment by xhudik** [comment]
good point @hmf  - new removed (where it was possible ;))

File: src/main/scala/ParameterUntupling.scala [hunk 1]
```diff
@@ -3,7 +3,7 @@
   */
 object ParameterUntupling {
```
Comment (Philippus) [inline src/main/scala/ParameterUntupling.scala:4]: still a `{` here, on purpose?

File: src/main/scala/PatternMatching.scala [hunk 2]
```diff
@@ -3,17 +3,15 @@
   */
 object PatternMatching {
```
Comment (Philippus) [inline src/main/scala/PatternMatching.scala:4]: here as well

File: src/main/scala/StructuralTypes.scala [hunk 3]
```diff
@@ -3,9 +3,8 @@
   */
 object StructuralTypes {
```
Comment (Philippus) [inline src/main/scala/StructuralTypes.scala:4]: and here

**Comment by xhudik** [comment]
thanks @Philippus , it seems automatic rewrites: `-indent -rewrite` didnt do a good job

**Comment by ShapelessCat** [comment]
Hi @xhudik,
Follow the Scala coding convention, I think we should replace all `def test: Unit =`s with `def test(): Unit =`s.

**Comment by xhudik** [comment]
Hi @ShapelessCat  - I'm not sure I;m following you. I didnt touch `def test: Unit =` (see [master branch](https://github.com/scala/scala3-example-project/blob/8fe5596980c1ccc8fb35f942a752e595b0677d16/src/main/scala/ContextFunctions.scala#L37) ).  There was not a version with `()`. I'm not scala style expert but I'd guess method: `def test: Unit =` is ok. 
Or, am I missing something?

**Comment by ShapelessCat** [comment]
@xhudik What I mean is since you are changing the style to use the new syntax, you can also help to fix some existing style issues. I don't mean those code weren't added by you.

You can ignore what I say. I can create a new PR for that.

**Comment by xhudik** [comment]
@ShapelessCat - indeed, you are right. I didnt get it first :). cool i change it - hope it is better now

File: src/main/scala/Conversion.scala [hunk 4]
```diff
@@ -3,22 +3,21 @@ import scala.language.implicitConversions
...
+object Conversion:
 
   case class IntWrapper(a: Int) extends AnyVal
   case class DoubleWrapper(b: Double) extends AnyVal
 
   def convert[T, U](x: T)(using converter: Conversion[T, U]): U = converter(x)
 
   given IntWrapperToDoubleWrapper: Conversion[IntWrapper, DoubleWrapper] = new Conversion[IntWrapper, DoubleWrapper] {
-    override def apply(i: IntWrapper): DoubleWrapper = new DoubleWrapper(i.a.toDouble)
+    override def apply(i: IntWrapper): DoubleWrapper =  DoubleWrapper(i.a.toDouble)
```
Comment (nicolasstucki) [inline src/main/scala/Conversion.scala:14]:
```suggestion
    override def apply(i: IntWrapper): DoubleWrapper = DoubleWrapper(i.a.toDouble)
```

File: src/main/scala/EnumTypes.scala [hunk 5]
```diff
@@ -21,21 +21,19 @@ object EnumTypes {
...
+    def calculateEarthWeightOnPlanets(earthWeight: Double) =
       val mass = earthWeight/Planet.Earth.surfaceGravity
-      for (p <- Planet.values)
+      for p <- Planet.values do
         println(s"Your weight on $p is ${p.surfaceWeight(mass)}")
-    }
 
+    println("Example 2:")
     calculateEarthWeightOnPlanets(80)
-  }
```
Comment (nicolasstucki) [inline src/main/scala/EnumTypes.scala:39]: For longer functions and classes it would be good to use the `end` marker.

```suggestion
  end test

```

**Review by nicolasstucki — COMMENTED** [review]
This PR needs to be rabased

File: src/main/scala/Conversion.scala [hunk 6]
```diff
@@ -27,9 +26,6 @@ object Conversion {
...
    */
 
-  def test: Unit = {
+  def test(): Unit =
     println(useConversion)
-    println(convert(new IntWrapper(42)))
-  }
-
-}
+    println(convert( IntWrapper(42)))
```
Comment (nicolasstucki) [inline src/main/scala/Conversion.scala:31]:
```suggestion
    println(convert(IntWrapper(42)))
```

**Comment by nicolasstucki** [comment]
Diff looks good but it needs to be rebased. I would also suggest squashing the commits.

File: src/main/scala/Main.scala [hunk 7]
```diff
@@ -1,33 +1,34 @@
-def main(args: Array[String]): Unit = {
+object Main:
```
Comment (nicolasstucki) [inline src/main/scala/Main.scala:1]: I missed this one in my previous review. We can update it to use the simplest for of `main` definition
```diff
- def main(args: Array[String]): Unit = {
+ @main def Main: Unit = {
```

File: src/main/scala/PatternMatching.scala [hunk 8]
```diff
@@ -77,7 +65,6 @@ object PatternMatching {
     import productPattern.*
     ("john", 42) match {
```
Comment (Philippus) [inline src/main/scala/PatternMatching.scala:66]: there is no closing } anymore for this match

File: src/main/scala/PatternMatching.scala [hunk 9]
```diff
@@ -94,7 +81,5 @@ object PatternMatching {
     "alice" match {
```
Comment (Philippus) [inline src/main/scala/PatternMatching.scala:81]: and also not for this one.

**Review by Philippus — DISMISSED** [review]

File: src/main/scala/Main.scala (hunk 7 above)
Comment (xhudik) [reply src/main/scala/Main.scala]: `@main`  didnt work for some reason :(

**Comment by xhudik** [comment]
@nicolasstucki  - I have rebased on master (forked repo and my lousy git knowledge  :)). Not sure whether squashing commits is a good idea (there is work of other people)

**Review by nicolasstucki — APPROVED** [review]

**Pull request merged**